File: packages/js/test/model-artifact.test.js
import assert from 'node:assert/strict';
import test from 'node:test';

import { defaultModelArtifact } from '../src/index.js';

test('bundled artifact parses into expected runtime shape', () => {
  assert.equal(defaultModelArtifact.modelId, 'baseline-v1');
  assert.equal(defaultModelArtifact.baseEngineId, 'spectral_ks_v1');
  assert.equal(defaultModelArtifact.inputDim, 23);
  assert.deepEqual(defaultModelArtifact.hiddenDims, [32, 32]);
  assert.equal(defaultModelArtifact.outputDim, 3);
  assert.equal(defaultModelArtifact.featureMean.length, defaultModelArtifact.inputDim);
  assert.equal(defaultModelArtifact.featureStd.length, defaultModelArtifact.inputDim);
  assert.equal(defaultModelArtifact.targetMean.length, defaultModelArtifact.outputDim);
  assert.equal(defaultModelArtifact.targetStd.length, defaultModelArtifact.outputDim);
  assert.ok(defaultModelArtifact.mixingParameters.learnedMixerBlend > 0);
});

File: packages/kotlin/src/commonTest/kotlin/io/github/rtarik/paintmixer/ModelArtifactTest.kt
package io.github.rtarik.paintmixer

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class ModelArtifactTest {

    @Test
    fun bundledArtifactParsesIntoExpectedRuntimeShape() {
        val runtime = DefaultModelArtifact.runtime

        assertEquals("baseline-v1", runtime.modelId)
        assertEquals("spectral_ks_v1", runtime.baseEngineId)
        assertEquals(23, runtime.inputDim)
        assertEquals(listOf(32, 32), runtime.hiddenDims)
        assertEquals(3, runtime.outputDim)
        assertEquals(runtime.inputDim, runtime.featureMean.size)
        assertEquals(runtime.inputDim, runtime.featureStd.size)
        assertEquals(runtime.outputDim, runtime.targetMean.size)
        assertEquals(runtime.outputDim, runtime.targetStd.size)
        assertTrue(runtime.mixingParameters.learnedMixerBlend > 0.0)
    }
}
